src/main/java/com/oracle/nosql/spring/data/core/NosqlTemplateBase.java

```diff
@@ -233,7 +386,7 @@ protected PutResult doPut(NosqlEntityInformation<?, ?> entityInformation,
                 putRes = nosqlClient.put(putReq);
             } catch (TableNotFoundException tnfe) {
                 if (entityInformation.isAutoCreateTable()) {
-                    doCreateTableIfNotExists(entityInformation);
+                    doCreateTable(entityInformation, getCreateTableDDL(entityInformation));
```

Comment:
Please make sure the table create/check is done at the entity initialization. If that is the case then this create table should be removed from here.

Reply:
This is an existing code. I don't know why create table is added as part of put.

Reply:
I might have missed it earlier.